### PR TITLE
[COD-254] fix: wrong number of calculated issues as part of the Sarif conversion

### PR DESCRIPTION
**Pull request by orkamara**
[COD-254]

- [ ] Ready for review

#### What does this PR do?
Fix an issue where ff there are multiple places, where same issue happens, they are lost during conversion to Sarif.

[COD-254]: https://snyksec.atlassian.net/browse/COD-254

### CODE REVIEW

File: src/sarif_converter.ts [hunk 1]
```diff
@@ -96,61 +96,65 @@ const getTools = (analysisResults: IAnalysisResult, suggestions: ISarifSuggestio
...
-    result[suggestionIndex] = {
-      ...suggestions[suggestionIndex],
-      ruleIndex,
-      rule,
-      level: severity,
-      id: suggestionId,
-      text: suggestion.message,
-    };
+    const allIssuesOfSuggestion: ISarifSuggestion[] = allIssuesBySuggestions[suggestionIndex];
+    allIssuesOfSuggestion.forEach(issue => {
```
Comment (miiila) [inline src/sarif_converter.ts:100]: NIT: I would vote more for `.map()` without side effects,
```
allIssues.push(...allIssuesOfSuggestion.map(issue => {...})
```

**Review by miiila — APPROVED** [review]
I quickly glanced and found nothing scary. Great job!

**Review by Spoor2709 — APPROVED** [review]
LGTM!

File: src/sarif_converter.ts (hunk 1 above)
Comment (orkamara) [reply src/sarif_converter.ts:100]: There are multiple changes I would like to do as part of this file - the code is really complex....

Comment (orkamara) [reply src/sarif_converter.ts:100]: As I'm short in time, and this is an important change, I would like us to do more improvements like that soon .... maybe next sprint?

**Pull request merged**